### PR TITLE
Add a IsClean() method to the etcdv3 datastore

### PR DESCRIPTION
**Pull request by robbrockbank**
## Description
Add a semi-private "IsClean()" method to the etcdv3 backend client.

If I end up implementing for KDD then I'll make it part of the backend API.

## Release Note

```release-note
None required
```

### CODE REVIEW

File: lib/backend/etcdv3/etcdv3.go [hunk 1]
```diff
@@ -395,7 +395,7 @@ func (c *etcdV3Client) EnsureInitialized() error {
 func (c *etcdV3Client) Clean() error {
 	log.Warning("Cleaning etcdv3 datastore of all Calico data")
 	_, err := c.etcdClient.Txn(context.Background()).If().Then(
-		clientv3.OpDelete("/calico", clientv3.WithPrefix()),
+		clientv3.OpDelete("/calico/", clientv3.WithPrefix()),
```
Comment (gunjan5) [inline lib/backend/etcdv3/etcdv3.go]: Why this change? It's prefix operation so should be fine, right?

File: lib/backend/etcdv3/etcdv3.go [hunk 2]
```diff
@@ -404,6 +404,21 @@ func (c *etcdV3Client) Clean() error {
...
 }
 
+// IsClean() returns true if there are no /calico/ prefixed entries in the
+// datastore.  This is not part of the exposed API, but is public to allow
+// direct consumers of the backend API to access this.
+func (c *etcdV3Client) IsClean() (bool, error) {
+	log.Debug("Calling Get on etcdv3 client")
+	resp, err := c.etcdClient.Get(context.Background(), "/calico/", clientv3.WithPrefix())
+	if err != nil {
+		log.WithError(err).Info("Error returned from etcdv3 client")
```
Comment (gunjan5) [inline lib/backend/etcdv3/etcdv3.go]: Shouldn't this be an error log level or a warning at least?

File: lib/backend/etcdv3/etcdv3.go (hunk 1 above)
Comment (robbrockbank) [reply lib/backend/etcdv3/etcdv3.go]: it's strictly more correct since this distinguishes from /calicov2/blah for example.

File: lib/backend/etcdv3/etcdv3.go (hunk 2 above)
Comment (robbrockbank) [reply lib/backend/etcdv3/etcdv3.go]: no - that's up to the end user if they want to log it as an error

generally lower down the stack you keep your levels more as debug or info.
